src/Castle.Core/DynamicProxy/Contributors/WrappedClassMembersCollector.cs

```diff
@@ -43,7 +43,7 @@ protected override MetaMethod GetMethodToGenerate(MethodInfo method, IProxyGener
 			}
 
 			var accepted = AcceptMethod(method, true, hook);
-			if (!accepted && !method.IsAbstract)
+			if (!accepted && Array.IndexOf(AllMethodsHook.skippedTypes, method.DeclaringType) >= 0)
```

Comment:
I suspect that the `!method.IsAbstract` check is no longer necessary, since we're going to forward to a target object (which wouldn't be instantiable if it had abstract methods).

Reply:
I added that additional filter because without it, DP will create forwarding implementations for each and every member available, including all `object` members. Letting `MinimalisticMethodGenerator` create implementations for two of those, `object.Finalize` and `object.MemberwiseCopy`, will cause PE verification to fail.

To be honest, I didn't analyse the cause for this in detail, instead I saw that `AllMethodsHook` (contrary to its name) has `SkippedTypes`, an exclusion list for members of certain types. I figured the PE verification might be why `SkippedTypes` exists in the first place.

If we remove that `Array.IndexOf` check...

* I'd need to figure out why PE verification fails, so it can be avoided.
* I supect we'd still need some exclusion list / criteria regarding certain members. In particular, I'm not sure how we'd deal with `object.Finalize`. If the target doesn't override that method, then ideally we would not generate a forwarding implementation, either, since this will negatively affect GC.
* There might be other special cases we'd need to account for: Do we need special logic for `object.Clone`, `object.Equals`, or is it always safe to forward these to the target?

Reply:
> `object.Finalize` and `object.MemberwiseCopy` will cause PE verification to fail

Ah, it's actually rather simple: those are `protected` and therefore not callable from the proxy. `ForwardingMethodGenerator` might have been written with interfaces in mind, it doesn't perform any accessibility checks.

So I think what I'll do is to remove that weird `SkippedTypes` check, and do an accessibility check instead. (`!method.IsPublic` should cover most cases, plus we might need some logic to recognize explicitly implemented interface methods and call them through the interface.)

Reply:
> `AllMethodsHook` (contrary to its name) has `SkippedTypes`

I think it was named that because it was the previous behaviour for all methods, most people don't need to intercept object methods.

> > `object.Finalize` and `object.MemberwiseCopy` will cause PE verification to fail
> 
> Ah, it's actually rather simple: those are `protected` and therefore not callable from the proxy

Yep, I don't recall ever having problems with those 2.